src/main/java/sb/service/OrderService.java

```diff
@@ -0,0 +1,182 @@
+        User user = userDAO.getByName(context.getAuthentication().getName())
+                .orElseThrow(() -> new NotFoundException("User is not found"));
+
+        if(user.getRole().equals("admin")) {
+            return orderDAO.getAll();
+        } else {
+            return orderDAO.getByUser(user.getId());
+        }
+    }
+
+
+    public Order get(int id) {
+        return orderDAO.get(id).orElseThrow(
+                () -> new NotFoundException("No such order")
+        );
+    }
+
+    public List<Order> getByUser(int id) {
+        return orderDAO.getByUser(id);
+    }
+
+
+    @Transactional
+    public Order create(OrderDTORequest orderDTORequest) {
+        SecurityContext context = SecurityContextHolder.getContext();
+        User user = userDAO.getByName(context.getAuthentication().getName())
+                .orElseThrow(() -> new NotFoundException("No such user")
+        );
+
+        List<LineItem> lineItems = orderDTORequest.getLineItems();
+
+        Optional<Product> product;
+        Optional<Stock> stock;
+
+        int orderId = orderDAO.post(user);
+
+        return save(orderId, orderDTORequest);
+    }
+
+
+    public Order updateStatus(String status, int orderId) {
+
+        orderDAO.get(orderId).orElseThrow(
+                () -> new NotFoundException("No such order")
+        );
+
+        if(status != null && (status.equals("shipped") || status.equals("unshipped"))){
+            orderDAO.update(status, orderId);
+        } else {
+            throw new CreationException("Wrong status");
+        }
+
+        return orderDAO.get(orderId).get();
+    }
+
+
+    @Transactional
+    public Order update(int orderId, OrderDTORequest orderDTORequest) {
+
+        if(orderDAO.get(orderId).isEmpty()) {
+            throw new NotFoundException("No such order");
+        }
+
+        if(orderDAO.get(orderId).get().getStatus().equals("shipped")) {
+            throw new NotFoundException("Can't update shipped order");
+        }
+
+        for (LineItem lineItem : lineItemDAO.getByOrder(orderId)) {
+
+            int productId = lineItem.getProductId();
+            Stock stock = stockDAO.get(productId).get();
+            int stockQuantity = stock.getQuantity();
+
+            stock.setQuantity(stockQuantity + lineItem.getQuantity());
+            stockDAO.update(stock);
+        }
+
+        lineItemDAO.deleteByOrder(orderId);
+
+        return save(orderId, orderDTORequest);
+    }
+
+
+    public void delete(int id) {
+        orderDAO.get(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+
+        lineItemDAO.deleteByOrder(id);
+        orderDAO.delete(id);
+    }
+
+    public Order save(int orderId, OrderDTORequest orderDTORequest) {
+        List<LineItem> lineItems = orderDTORequest.getLineItems();
+
+        Optional<Product> product;
+        Optional<Stock> stock;
+
+        List<LineItem> lineItemSet = new ArrayList<>();
```

Comment:
`List<LineItem> lineItemSet` looks a bit confusing...

Reply:
fix: rename into lineItemOrigin

Reply:
What is lineItemOrigin? How about simple name it lineItems?

Reply:
There are two lineItems, first: sent lineitems from orderDTO (can be copies), second: lineItemSet, which has no dublicates. It was List, now it is Set. rows 140 - .. logic of not saving doubles